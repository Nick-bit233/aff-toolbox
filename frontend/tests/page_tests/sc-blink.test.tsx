import { render, fireEvent, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { echoServer } from "../mock";
import ToolPage from "@/pages/tools/sc-blink";

jest.mock("next/router", () => jest.requireActual("next-router-mock"));
jest.mock("react-redux");

beforeAll(() => echoServer.listen());
afterEach(() => echoServer.resetHandlers());
afterAll(() => echoServer.close());

test("echo test", async () => {
  const user = userEvent.setup();
  // @ts-ignore
  // delete window.location;
  // @ts-ignore
  // window.location = new URL("https://www.example.com/tools/sc-blink");
  render(<ToolPage />);
  await user.type(
    screen.getByRole("textbox", { name: "input.params.start" }),
    "1"
  );
  expect(
    (
      screen.getByRole("textbox", {
        name: "input.params.start",
      }) as HTMLInputElement
    ).value
  ).toBe("1");
});
