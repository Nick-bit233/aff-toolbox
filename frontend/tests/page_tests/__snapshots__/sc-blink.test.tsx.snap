// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form test all 1`] = `"{"params":{"sc_x":2,"count":10,"stop":1000,"start":0}}"`;

exports[`form test all required 1`] = `"{"params":{"count":10,"stop":1000,"start":0}}"`;

exports[`form test required missing 1`] = `"["params"]"`;

exports[`form test required missing 2`] = `"["params"]"`;

exports[`form test required missing 3`] = `"["params"]"`;
